### ISSUE TITLE
Audio data cannot be converted automatically from int8 to 16-bit int format

### ISSUE THREAD

**Issue by Ram-Pasupula** [gradio-app/gradio#7099]
### Describe the bug

It appears that the processing_utils library lacks support for int8 when using the convert_to_16_bit_wav(data) function. 

Code:  
gr.Interface(
            fn=transcribe,
            inputs=[gr.Audio(type="filepath", label="Audio File"),
                    gr.Dropdown(
                    choices=["es", "en"],  label="output Language",  value="en")],
            outputs=gr.Textbox(value="", show_copy_button=True))

code: Gradio -processing_utils -
def convert_to_16_bit_wav(data):
    # Based on: https://docs.scipy.org/doc/scipy/reference/generated/scipy.io.wavfile.write.html
    warning = "Trying to convert audio automatically from {} to 16-bit int format."
    if data.dtype in [np.float64, np.float32, np.float16]:
        warnings.warn(warning.format(data.dtype))
        data = data / np.abs(data).max()
        data = data * 32767
        data = data.astype(np.int16)
    elif data.dtype == np.int32:
        warnings.warn(warning.format(data.dtype))
        data = data / 65538
        data = data.astype(np.int16)
    elif data.dtype == np.int16:
        pass
    elif data.dtype == np.uint16:
        warnings.warn(warning.format(data.dtype))
        data = data - 32768
        data = data.astype(np.int16)
    elif data.dtype == np.uint8:
        warnings.warn(warning.format(data.dtype))
        data = data * 257 - 32768
        data = data.astype(np.int16)
    else:
        raise ValueError(
            "Audio data cannot be converted automatically from "
            f"{data.dtype} to 16-bit int format."
        )
    return data

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

```python
import gradio as gr

```

### Logs

```shell
Audio data cannot be converted automatically from int8 to 16-bit int format
```

### System Info

```shell
gradio version 4.15.0
```

### Severity

Blocking usage of gradio

**Comment by abidlabs** [comment]
It looks like it indeed. Would you be able to provide the audio file you are using so that we can repro?

**Comment by Ram-Pasupula** [comment]
Thank you for your prompt response. Unfortunately, I can't share the file as it contains Personally Identifiable Information (PII) data in production. However, you can convert int16 to int8 using the following code:
if audio_array.dtype == np.int16:
    audio_array = audio_array.astype(np.int8)


The workaround I have implemented is:

gr.Audio(type="numpy", label="Audio File")
 
def transcribe(audio_data, lang):
    audio_file_format, audio_array = audio_data
    ....
    wav_file_path = "/tmp/input_audio.wav"
    
    if audio_array.dtype == np.int8:
        audio_array = audio_array.astype(np.int16) 
    
    import soundfile as sf    
    sf.write(wav_file_path, audio_array, audio_file_format, subtype='PCM_16')
    
    # Create a tuple for file upload
    files = {'file': ('input_audio.wav', open(wav_file_path, 'rb'))}
    
    print(f"lang: {lang}")
    
    params = {"task": "transcribe",
              "lang": lang,
              "output": "txt",
    }
    
    response = requests.post(BACKEND_API_URL, files=files, params=params)
   .....

**Comment by abidlabs** [comment]
That works thanks, we’ll look into this when we get a chance. If you have a fix in mind, feel free to open a PR and we can review

**Comment by Ram-Pasupula** [comment]
Opened a PR, fix#7099 #7112. Thanks.
